services/dashboard/views/apps/app/appmaster.js

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed under the Apache License, Version 2.0
+ * See accompanying LICENSE file.
+ */
+'use strict';
+angular.module('dashboard.apps.appmaster', ['directive.visgraph', 'dashboard.streamingdag'])
+
+  .config(['$routeProvider', function ($routeProvider) {
+    $routeProvider
+      .when('/apps/app/:id', {
+        label: 'Application ',
+        templateUrl: 'views/apps/app/appmaster.html',
+        controller: 'AppMasterCtrl'
+      });
+  }])
+
+  .controller('AppMasterCtrl',
+  ['$scope', '$routeParams', 'breadcrumbs', 'restapi', 'StreamingService', 'visdagUtil', 'StreamingDag',
+    function ($scope, $routeParams, breadcrumbs, restapi, StreamingService, visdagUtil, StreamingDag) {
+      $scope.app = {id: $routeParams.id};
+      breadcrumbs.options = {'Application ': 'Application ' + $scope.app.id};
+
+      $scope.streamingDag = null;
+      restapi.subscribe('/appmaster/' + $scope.app.id + '?detail=true', $scope,
+        function (data) {
```

Comment:
Add reference here to the data type GeneralAppMasterDataDetail. Something like:
 //TODO  datatype: GeneralAppMasterDataDetail. Try and do serde using scalajs See #458